frontend/src/main/java/olumofe/raphael/frontend/CarController.java

```diff
@@ -0,0 +1,63 @@
+package olumofe.raphael.frontend;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+import starter.project.domain.Car;
+
+import java.util.*;
+
+
+@RestController
+public class CarController {
+
+    private List<Car> carNewList = new ArrayList<>();
+    Map<String, Car> map = new HashMap<>();
+
+    @PostMapping(path = "/car", consumes = {"application/json"})
+    public List<Car> addCar(@RequestBody Car car) {
+        carNewList.add(car);
+        return carNewList;
+    }
+
+    @GetMapping(path = "/car/{carId}")
+    public Car getCar(@PathVariable("carId") int carId) {
+
+        for (Car car : carNewList) {
+            if (car.getId() == (carId)) {
+                return car;
+            } else {throw new ResponseStatusException(
+                    HttpStatus.NOT_FOUND, "entity not found");
+            }
+        }
+        return null;
```

Comment:
We should not return a `null` in this case, probably a 404. I would remove thois `else` branch:
```java
else {throw new ResponseStatusException(
                    HttpStatus.NOT_FOUND, "entity not found");
            }
```
And throw the exception at this point inconditionally

Reply:
resolved